arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedPageIterator.java

```diff
@@ -312,7 +319,7 @@ public int nextBatchFixedLengthDecimal(
     if (actualBatchSize <= 0) {
       return 0;
     }
-    if (eagerDecodeDictionary) {
+    if (dictionaryDecodeMode == DictionaryDecodeMode.EAGER) {
```

Comment:
Coverage shows that this path isn't taken either.

Reply:
Looks like `DictionaryDecimalBinaryAccessor` is also not used by tests. That's probably related. Maybe decimals are not getting dictionary encoded?